### PR TITLE
p2p: add PeerManager.Advertise()

### PR DESCRIPTION
**Pull request by erikgrinaker**
Adds a naïve `PeerManager.Advertise()` method that the new PEX reactor can use to fetch addresses to advertise, as well as some other `FIXME`s on address advertisement.

### CODE REVIEW

File: p2p/peer.go [hunk 1]
```diff
@@ -492,6 +492,34 @@ func (m *PeerManager) Add(address PeerAddress) error {
 	return nil
 }
 
+// Advertise returns a list of peer addresses to advertise to a peer.
+//
+// FIXME: We currently just pass all addresses we have, which is very naïve. We
+// should e.g. only send addresses that the peer can actually reach (by
+// resolving the addresses into endpoints and making sure any private IP
+// addresses are on the same network as the remote peer endpoint). However, this
+// would require resolving endpoints either here (too slow) or generally in the
```
Comment (melekes) [inline p2p/peer.go:501]: would it make sense to only resolve the first endpoint? if not successful, we continue resolving 2nd, 3rd ... N endpoints

Comment (erikgrinaker) [reply p2p/peer.go:501]: Hm? Resolving an address will always return all endpoints, that's how DNS works. The problem is that we may want to advertise e.g. 100 addresses, and resolving all 100 addresses into endpoints on demand would be slow. We may want to have the peer manager resolve the endpoints when it's first given the address and then periodically refresh it.

Comment (melekes) [reply p2p/peer.go:501]: > We may want to have the peer manager resolve the endpoints when it's first given the address and then periodically refresh it.

that sounds fine to me

**Review by alexanderbez — APPROVED** [review]

**Comment by alexanderbez** [comment]
I think this suffices for now as a naive approach.

**Pull request merged**